scripts/database.sql

```diff
@@ -179,7 +180,9 @@ CREATE TABLE Lanefile (
 	-- this is some simple change tracking
 	-- on every change a new Lanefile is stored, with the old contents and original_id referencing the real Lanefile
 	original_id    int        NULL REFERENCES Lanefile (id),
-	changed_date   timestamp  NULL -- the date the change was made	
+	changed_date   timestamp  NULL, -- the date the change was made
+
+	required_roles text NULL DEFAULT NULL	-- Roles required to access the Lanefile
```

Comment:
When a user creates a new file in a lane, it copies over the `required _roles` from the lane it was created from (so the user can still see it). For a LaneFile that has lesser permissions than what the current user has in the lane they are in, they would be unauthorized. Someone with role access or an admin does need to go in and add the user role to it. This is indented behavior, because if you want to have a permission system for who can see and access a file, you should be explicit about it and add them in. This goes for the file history too. When you change the required roles for the file, users can go to the file history and see there were changes, but can't access them (since they didn't have access to them at that time). Also for admins, nothing changes. They get full access to every file or lane regardless of the roles in those fields.

For LaneFiles, they only need to do this once, since the `required_roles` for the LaneFile are separate from the lane. So I would think for shared resources this would be fine. 

I